packages/cli/test/integration/credentials.ee.test.ts

```diff
@@ -0,0 +1,270 @@
+import express from 'express';
+import { UserSettings } from 'n8n-core';
+import { Db } from '../../src';
+import { randomName, randomString } from './shared/random';
+import config from '../../config';
+import * as utils from './shared/utils';
+import type { CredentialPayload, SaveCredentialFunction } from './shared/types';
+import type { Role } from '../../src/databases/entities/Role';
+import type { User } from '../../src/databases/entities/User';
+import * as testDb from './shared/testDb';
+import { RESPONSE_ERROR_MESSAGES } from '../../src/constants';
+import { CredentialsEntity } from '../../src/databases/entities/CredentialsEntity';
+
+jest.mock('../../src/telemetry');
+
+let app: express.Application;
+let testDbName = '';
+let globalOwnerRole: Role;
+let globalMemberRole: Role;
+let credentialOwnerRole: Role;
+let saveCredential: SaveCredentialFunction;
+
+beforeAll(async () => {
+	app = await utils.initTestServer({
+		endpointGroups: ['credentials'],
+		applyAuth: true,
+	});
+	const initResult = await testDb.init();
+	testDbName = initResult.testDbName;
+
+	utils.initConfigFile();
+
+	globalOwnerRole = await testDb.getGlobalOwnerRole();
+	globalMemberRole = await testDb.getGlobalMemberRole();
+	credentialOwnerRole = await testDb.getCredentialOwnerRole();
+	saveCredential = affixRoleToSaveCredential(credentialOwnerRole);
+
+	utils.initTestLogger();
+	utils.initTestTelemetry();
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['User', 'SharedCredentials', 'Credentials'], testDbName);
+	config.set('deployment.paid', true);
+});
+
+afterAll(async () => {
+	await testDb.terminate(testDbName);
+});
+
+// ----------------------------------------
+// dynamic router switching
+// ----------------------------------------
+
+test('dynamic router switching', async () => {
```

Comment:
hmmmm I think this would be too specific, in the end I test if the router can be switched dynamically based on the config. I just use the status codes as my test value

Reply:
`should dynamically switch routers`?